processing/src/main/java/org/apache/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -608,7 +608,7 @@ IncrementalIndexRowResult toIncrementalIndexRow(InputRow row)
         Object dimsKey = null;
         try {
           final EncodedKeyComponent<?> encodedKeyComponent
-              = indexer.processRowValsToUnsortedEncodedKeyComponent(row.getRaw(dimension), true);
+              = indexer.processRowValsToUnsortedEncodedKeyComponent(row.getRaw(dimension), true, dimension);
```

Comment:
hmm, do we really need to pass this as an argument for every value of every row? maybe it would be better to just initialize the dimension indexers with their name instead.

Alternatively, we are already catching `ParseException`, why not add a way to decorate it with an additional column name.

I guess i'm just being cautious about small overhead of additional arguments adding up since this is a pretty hot function

Reply:
@clintropolis I think i could do that. Lemme get this current PR passing. then i'll attempt that

Reply:
@clintropolis i was able to initialize the dimensionIndexers with the name

Reply:
cool, that seems better. 

thinking a bit more on it i think having a way to decorate an existing `ParseException` with a column name might actually have been a little cleaner and even less disruptive since we are already catching them in a place with the column name, https://github.com/apache/druid/blob/master/processing/src/main/java/org/apache/druid/segment/incremental/IncrementalIndex.java#L616 without being dependent on dimension indexer implementation, but this way seems ok too